contracts/src/v0.8/vrf/dev/VRFCoordinatorV2_5.sol

```diff
@@ -176,9 +178,16 @@ contract VRFCoordinatorV2_5 is VRF, SubscriptionAPI, IVRFCoordinatorV2Plus {
     if (fallbackWeiPerUnitLink <= 0) {
       revert InvalidLinkWeiPrice(fallbackWeiPerUnitLink);
     }
-    if (fulfillmentFlatFeeNativePPM > 0 && fulfillmentFlatFeeLinkDiscountPPM >= fulfillmentFlatFeeNativePPM) {
+    if (
+      fulfillmentFlatFeeLinkDiscountPPM > fulfillmentFlatFeeNativePPM ||
+      (fulfillmentFlatFeeNativePPM > 0 && fulfillmentFlatFeeLinkDiscountPPM == fulfillmentFlatFeeNativePPM)
```

Comment:
Actually what's wrong with just checking `fulfillmentFlatFeeLinkDiscountPPM >= fulfillmentFlatFeeNativePPM`? Like `fulfillmentFlatFeeLinkDiscountPPM` can be equals to `fulfillmentFlatFeeNativePPM` right? One scenario could be we want to impose a flat fee for native, but no flat fee for link.

Or actually why don't we just have 1 flat fee for native and 1 flat fee for link. Conceptually we want to give a discount for using link tokens and peg it to native. But it's not like we have individual setter functions that can set just the flat fee link discount. We would have to set both the flat fee native and flat fee link discount in a single setting anyways. Not sure which approach will be less confusing.

Reply:
If we check for `fulfillmentFlatFeeLinkDiscountPPM >= fulfillmentFlatFeeNativePPM`, then when both fulfillmentFlatFeeLinkDiscountPPM == 0 and fulfillmentFlatFeeNativePPM == 0, like what we plan for the initial launch in most chains (to have precentagePremium instead of flat fee), the check would fail and revert.

Reply:
I don't get why `fulfillmentFlatFeeLinkDiscountPPM` has to be **strictly** less than `fulfillmentFlatFeeNativePPM`

Reply:
The original logic was 
```
    if (fulfillmentFlatFeeNativePPM > 0 && fulfillmentFlatFeeLinkDiscountPPM >= fulfillmentFlatFeeNativePPM) {
      revert LinkDiscountTooHigh(fulfillmentFlatFeeLinkDiscountPPM, fulfillmentFlatFeeNativePPM);
    }
```

We want to cover the zero scenario as well: `fulfillmentFlatFeeNativePPM >= 0`, but then both fulfillmentFlatFeeLinkDiscountPPM and fulfillmentFlatFeeNativePPM being 0 is a valid scenario and should not revert, which is what this condition checks:

```
    if(fulfillmentFlatFeeLinkDiscountPPM > fulfillmentFlatFeeNativePPM ||
      (fulfillmentFlatFeeNativePPM > 0 && fulfillmentFlatFeeLinkDiscountPPM == fulfillmentFlatFeeNativePPM))
```

Reply:
The original logic was already checking for  fulfillmentFlatFeeLinkDiscountPPM to be strictly less than fulfillmentFlatFeeNativePPM
```
if (fulfillmentFlatFeeNativePPM > 0 && fulfillmentFlatFeeLinkDiscountPPM >= fulfillmentFlatFeeNativePPM) {
```

As part of this ticket, we needed to add this check for the scenario of fulfillmentFlatFeeNativePPM being 0 as well. However, when both fulfillmentFlatFeeLinkDiscountPPM and fulfillmentFlatFeeNativePPM are 0, it is a valid scenario and we should not revert.

```
if (
      fulfillmentFlatFeeLinkDiscountPPM >= fulfillmentFlatFeeNativePPM &&
      !(fulfillmentFlatFeeNativePPM == 0 && fulfillmentFlatFeeLinkDiscountPPM == 0)
    ) {
```

Reply:
As discussed offline with Todor and @leeyikjiun, we can set fulfillmentFlatFeeLinkDiscountPPM equal to fulfillmentFlatFeeNativePPM.